lib/chef/knife/base_vsphere_command.rb

```diff
@@ -223,10 +223,22 @@ def find_folder(folderName)
         base_entity
       end
 
-      def find_network(networkName)
+      def find_network(networkName, dvswitch = nil)
+        switch_uuid = true if dvswitch =~ /^([0-9a-f]{2}[ -]{0,1})*$/
```

Comment:
Could you please extract the regexp to a constant at the top of the class so that what we get in the method is a bit more clear to the reader? (e.g. `UUID`, `IS_UUID`, `LOOKS_LIKE_A_FINE_UUID`)

I'm not positive this regex effectively filters out non-UUIDs. The `*` at the end means none of it has to match, so a blank will match, and the `{0,1}` (aka `?`) means that we could match on character strings. As will any network name that contains `[0-9a-f]` and has an even number of characters:
```
irb(main):007:0> UUID=%r{^([0-9a-f]{2}[ -]{0,1})*$}
=> /^([0-9a-f]{2}[ -]{0,1})*$/
irb(main):008:0> '' =~ UUID
=> 0
irb(main):009:0> 'deadbeef' =~ UUID
=> 0
irb(main):010:0> nil =~ UUID
=> nil
irb(main):011:0> '' =~ UUID
=> 0
```

Add to that, we're also imposing on the user to enter the uuid in lowercase.

It seems like it would be hard to detect something that's a UUID and not just a network name that looks uuid-ish.

What would you think about checking both `uuid` and `name` without worrying about it being a uuid or not? The corner case would be someone who named their dswitch with a uuid that's different from the actual uuid, but happens to correspond to an actual uuid.

That would simplify the `if/return if return` block methinks.